packages/stencil/components/pagination/src/components/osds-pagination/osds-pagination.tsx

```diff
@@ -0,0 +1,200 @@
+import { Component, Element, Event, EventEmitter, Host, Method, Prop, Watch, h, Fragment } from '@stencil/core';
+import {
+  OdsLogger,
+  OdsPagination,
+  OdsPaginationController,
+  odsPaginationDefaultAttributes,
+  OdsPaginationEvents,
+  OdsPaginationMethods,
+  OdsPaginationCurrentChangeEventDetail,
+  OdsIconSize,
+  OdsIconName,
+  OdsButtonSize,
+  OdsButtonVariant,
+  OdsTextSize,
+} from '@ovhcloud/ods-core';
+import { HTMLStencilElement } from '@stencil/core/internal';
+import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';
+import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
+
+/**
+ * @slot (placeholder) - Pagination placeholder
+ */
+@Component({
+  tag: 'osds-pagination',
+  styleUrl: 'osds-pagination.scss',
+  shadow: true,
+})
+export class OsdsPagination implements OdsPagination<OdsStencilMethods<OdsPaginationMethods>, OdsStencilEvents<OdsPaginationEvents>> {
+  private logger = new OdsLogger('OsdsPagination');
+  controller: OdsPaginationController = new OdsPaginationController(this);
+
+  @Element() el!: HTMLStencilElement;
+
+  /** @see OdsPaginationAttributes.current */
+  @Prop({ reflect: true, mutable: true }) current: number = odsPaginationDefaultAttributes.current;
+
+  /** @see OdsPaginationAttributes.defaultCurrent */
+  @Prop({ reflect: true }) defaultCurrent?: number = odsPaginationDefaultAttributes.defaultCurrent;
```

Comment:
it think it is useless and overkill as you can take the same feature with the `current` property:
- no current prop => set the default one from odsPaginationDefaultAttributes.current
- current="1" => use it
- no need to set it inside `componentWillLoad`
- if user want to set the default page to `2`, so he set current="2"